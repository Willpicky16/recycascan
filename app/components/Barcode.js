import React, { Component } from 'react';
import { StyleSheet, Text, View, Dimensions } from 'react-native';
import Camera from 'react-native-camera';
import axios from 'axios';

const ROOT = 'https://gentle-escarpment-88131.herokuapp.com/api/products';

export default class Barcode extends Component {
  scanBarcode (data) {
    getName(data.data);
  }
  render () {
    return (
      <Camera
        ref={(cam) => {this.camera = cam;}}
        style={styles.preview}
        onBarCodeRead={code => this.scanBarcode(code)}
        aspect={Camera.constants.Aspect.fill}>
      </Camera>
    )
  }
}

function getName (EAN) {
  axios
    .get(`${ROOT}/${EAN}`)
    .then(function scanBarcode (data) {
      alert(`${data.data.products[0].name} (${EAN}) Packaging: ${data.data.products[0].packaging}`)
    })
    .catch(function (err) {
      alert(EAN)
    })
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  preview: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width
  }
});
