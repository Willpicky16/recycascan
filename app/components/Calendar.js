import React, { Component } from "react";
import { StyleSheet, Text, View, ScrollView, AsyncStorage, Image } from "react-native";
import axios from 'axios';
import moment from 'moment';

let day = moment().format('D')
let month = moment().format('MMMM');
let year = moment().format('YYYY');
console.log(`${day} ${month} ${year}`);

export default class Calendar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collections: [],
      userDetails: {}
    };
    this.getCollections = this.getCollections.bind(this);
  }

  componentDidMount() {
    AsyncStorage.getItem('userDetails', (err, result) => {
      let val = JSON.parse(result);
      this.setState({
        userDetails: val
      });
      this.getCollections(val.council);
    });
  }

  getCollections (council) {
    axios
      .get(`https://vast-eyrie-43528.herokuapp.com/api/collections?council=${council}`)
      .then((res) => {
        this.setState({
          collections: res.data.collections
        });
        console.log(this.state.collections[5].bins[2]);
      })
      .catch((err) => {
        alert(err);
      });
  }

  render() {
    if (this.state.userDetails === null) return (
      <View style={styles.containerError}>
        <Text style={styles.titleError}>Please enter a postcode</Text>
      </View>
    )
    return (
      <ScrollView>
        <View style={styles.container}>
            {this.state.collections.map((collection, key) => {
                return (
                  <View key={key}>
                    <Text style={styles.date}>
                      {collection.day.toString()} {collection.month.toString()} {collection.year.toString()} ({collection.council.toString()})
                    </Text>
                    {collection.bins.map((bin, i) => {
                      return (
                        <View style={styles.imageView}>
                          <Image style={styles.binImage} source={{ uri: bin }}/>
                        </View>
                      )
                    })}
                  </View>
                )
              })}
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  containerError: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  date: {
    paddingTop: 50,
    paddingBottom: 10,
    fontSize: 20
  },
  titleError: {
    fontSize: 20,
    color: 'red'
  },
  bins: {
    color: 'blue'
  },
  binImage: {
    height: 100,
    width: 65
  },
  imageView: {
    flexDirection: 'row'
  }
});
