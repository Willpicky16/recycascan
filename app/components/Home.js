import React, { Component } from 'react';
import { StyleSheet, Text, View, Button, AsyncStorage, Image, Vibration, TouchableNativeFeedback } from 'react-native';
import { Form, InputField } from 'react-native-form-generator';
import { Restart } from 'react-native-restart';
// import { postcodeToCouncil } from '../helpers/postcodeToCouncil';
import axios from 'axios';

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      postcode: '',
      userDetails: {
        council: '',
        postcode: ''
      }
    }
    this.postcodeToCouncil = this.postcodeToCouncil.bind(this);
  }

  handleFormChange(data) {
    this.setState({
      postcode: data.postcode
    });
  }

  onButtonPress() {
    if (this.state.postcode === '') return alert('Please enter a postcode');
    this.postcodeToCouncil(this.state.postcode);
    Vibration.vibrate();
  }

  onButtonClear() {
    AsyncStorage.clear();
    Vibration.vibrate();
    Restart();
  }

  postcodeToCouncil (str) {
    if (str.length > 7 || str.length < 5) alert('Invalid postcode');
    let code;
    if (str.match(/\s/)) {
      code = str.match(/^\S*/)[0].toUpperCase();
    } else {
      code = str.length > 5 ? str.slice(0, 3).toUpperCase() : str.slice(0, 2).toUpperCase();
    }
    axios
      .get('https://vast-eyrie-43528.herokuapp.com/api/postcodes')
      .then(function (res) {
        let value = {
          council: res.data.postcode.find(key => key.postcode === code).council,
          postcode: str.toUpperCase()
          };
        AsyncStorage.setItem('userDetails', JSON.stringify(value));
        // alert(`Postcode ${str} submitted`);
        Restart();
      })
      .catch(function (err) {
        alert(err);
      });
  }
  watchID: ?number = null;

  componentDidMount () {
    AsyncStorage.getItem('userDetails', (err, result) => {
      let val = JSON.parse(result)
      this.setState({
        userDetails: val
      });
    });

    navigator.geolocation.getCurrentPosition(
      (position) => {
        console.log(position);
      },
      (error) => console.log(JSON.stringify(error)),
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
    );
    this.watchID = navigator.geolocation.watchPosition((position) => {
      console.log(position)
    });
  }

  render () {
    if (this.state.userDetails === null) return (
      <View style={styles.container}>
        <Image style={styles.logo} source={require('../images/recycascan.png')}/>
        <Text style={styles.title}>Enter your postcode to begin</Text>
        <Form label="Add new product form" ref="form" onChange={this.handleFormChange.bind(this)}>
          <InputField ref="postcode" placeholder="Postcode" helpText="                           "/>
        </Form>
        <TouchableNativeFeedback
            onPress={this.onButtonPress.bind(this)}
            background={TouchableNativeFeedback.SelectableBackground()}
            accessibilityLabel="Submit">
          <View style={styles.button} >
            <Text style={styles.buttonText}>SUBMIT</Text>
          </View>
        </TouchableNativeFeedback>
      </View>
    )
    return (
      <View style={styles.container}>
        <Image style={styles.logo} source={require('../images/recycascan.png')}/>
        <Text style={styles.title}> My postcode</Text>
        <Text style={styles.postcode}>{this.state.userDetails.postcode}</Text>
        <TouchableNativeFeedback
            onPress={this.onButtonClear.bind(this)}
            background={TouchableNativeFeedback.SelectableBackground()}
            accessibilityLabel="Edit">
          <View style={styles.button} >
            <Text style={styles.buttonText}>EDIT</Text>
          </View>
        </TouchableNativeFeedback>
      </View>
    )
  }
}

// {this.state.userDetails.council}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: '#A3D860'
  },
  userDetailsTitle: {
    paddingBottom: 20,
    fontSize: 20,
    color: '#004400'
  },
  postcode: {
    paddingBottom: 40,
    fontSize: 50,
    color: '#004400',
    fontWeight: '300'
  },
  title: {
    paddingTop: 60,
    paddingBottom: 10,
    fontSize: 20,
    color: '#004400'
  },
  logo: {
    width: 200,
    height: 100
  },
  button: { 
    height: 40,
    width: 200,
    borderRadius: 5,
    backgroundColor: '#004400'
  },
  buttonText: {
    color: '#fff', 
    padding: 10, 
    textAlign: "center"
  }
});