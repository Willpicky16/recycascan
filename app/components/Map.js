import React, { Component } from 'react';
import { StyleSheet, Text, View, AsyncStorage } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import axios from 'axios';

export default class Map extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userDetails: {
        postcode: 'M11 3JA'
      },
      region: {
        latitude: 53.4808,
        longitude: -2.2426,
        latitudeDelta: 0.5,
        longitudeDelta: 0.5
      },
      centres: []
    }
  }
  componentDidMount() {
    AsyncStorage.getItem('userDetails', (err, result) => {
      let val = JSON.parse(result);
      if (val !== null) this.setState({
        userDetails: val
      }, () => {
        axios
          .get(`http://postcodes.io/postcodes/${this.state.userDetails.postcode}`)
          .then((res) => {
            this.setState({
              region: {
                latitude: res.data.result.latitude,
                longitude: res.data.result.longitude,
                latitudeDelta: 0.0922,
                longitudeDelta: 0.0421
              }
            })
          })
          .catch((err) => {
            console.log(err)
          });
      });
    });
    axios
      .get('https://vast-eyrie-43528.herokuapp.com/api/recyclingcentres')
      .then((res) => {
        this.setState ({
          centres: res.data.recyclingcentres
        })
      })
      .catch((err) => {
        console.log(err)
      });
  }

  render() {
    console.log(this.state);
    return (
      <View style={styles.container}>
        <MapView
          style={styles.map}
          region={this.state.region}
        >
          {this.state.centres.map((centre, i) => {
            return (
              <MapView.Marker
                key={i}
                coordinate={{ latitude: centre.latitude, longitude: centre.longitude }}
                title={centre.name}
                image={require('../images/pin.png')}
                pinColor={'#00c60e'}
              />
            );
          })}
        </MapView>
      </View >
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
});
