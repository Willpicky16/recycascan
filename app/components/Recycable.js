import React, { Component } from "react";
import { StyleSheet, Text, View, AsyncStorage } from "react-native";
import axios from "axios";

const ROOT = 'https://world.openfoodfacts.org/api/v0/product/';
const BINROOT = 'https://vast-eyrie-43528.herokuapp.com/api/bins';

export default class Recycable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userDetails: {},
      product: {},
      bin: '',
      loading: true
    };
  }
  componentDidMount() {
    AsyncStorage.getItem('userDetails', (err, result) => {
      let val = JSON.parse(result);
      this.setState({
        userDetails: val
      });
    });
    axios
      .get(`${ROOT}/${this.props.navigation.state.params.code}`)
      .then(res => {
        axios
          .get(`${BINROOT}?packaging=${res.data.product.packaging}&council=${this.state.userDetails.council}`)
          .then((res) => {
            this.setState({
              bin: res.data.bins[0].bin,
              loading: false
            })
            console.log(this.state);
          })
          .catch((err) => {
            console.log(err);
          })
        this.setState({
          product: res.data.product
        });
      })
      .catch(err => {
        console.log(err);
        this.props.navigation.navigate('ProductSubmit', {code: this.props.navigation.state.params.code});
      });
  }
  render() {
    if (this.state.loading === true)
      return (
        <View style={styles.container}>
          <Text style={styles.loading}>loading....</Text>
        </View>
      );
    return (
      <View style={styles.container}>
        <Text style={styles.subtitle}>{this.state.product.product_name} ({this.state.product.code})</Text>
        <Text style={styles.subtitle}>
          Put in {this.state.bin} bin!
        </Text>
      </View>
    );
  }
}

// function getBinColour () {
//   axios
//     .get(`${BINROOT}?packaging=${res.data.product.packaging}&council=${this.state.council}`)
//     .then((res) => {
//       console.log(res);
//     })
//     .catch((err) => {
//       console.log(err);
//     })
// }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  title: {
    color: "blue",
    fontSize: 25
  },
  subtitle: {
    fontSize: 20
  },
  loading: {
    color: "red",
    fontSize: 30
  }
});
