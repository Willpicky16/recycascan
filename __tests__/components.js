import 'react-native';
import React from 'react';

import Home from '../app/components/Home';
import Barcode from '../app/components/Barcode';
import Calendar from '../app/components/Calendar';
import Map from '../app/components/Map';
import Recycable from '../app/components/Recycable';

import renderer from 'react-test-renderer';

test('home page renders correctly', () => {
  const tree = renderer.create(
    <Home />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('barcode page renders correctly', () => {
  const tree = renderer.create(
    <Barcode />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('calendar page renders correctly', () => {
  const tree = renderer.create(
    <Calendar />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('map page renders correctly', () => {
  const tree = renderer.create(
    <Map />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('recycable page renders correctly', () => {
  const tree = renderer.create(
    <Recycable />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
