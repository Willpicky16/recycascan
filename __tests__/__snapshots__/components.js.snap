// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`barcode page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Barcode
  </Text>
</View>
`;

exports[`calendar page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Calendar
  </Text>
</View>
`;

exports[`home page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home
  </Text>
</View>
`;

exports[`map page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Map
  </Text>
</View>
`;

exports[`recycable page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Recycable
  </Text>
</View>
`;
